inception-project/inception
# Feature/914 gazeteer support for string matching recommender

**What's in the PR**
- [x] Add ability to imported / download gazeteers to the String recommender traits UI
- [x] Created a new DB table to manage gazeteers
- [x] Implemented unit tests for the new GazeteerService
- [x] Added documentation on gazeteers
- [x] Allow RecommenderEditorPanel to scroll its content to allow for large traits editors
- [x] Expanded scope of the try/catch clauses in PredictionTask and TrainingTask to cover any exceptions that might be thrown during recommender initialization
- [x] Handle importing two Gazeteers by the same name
- [x] Handle Gazeteer containing multiple labels for the same text
- [x] Remove `setOutputMarkupPlaceholder` on `upload` field 
- [x] Support Gazeteers in project export/import

**How to test manually**
* Create a project with a string recommender
* Import a text
* Create a simple gazeteer file containing label/word pairs separated by tabs for some of the words (or multi-words) in the text
* Import the gazeteer into the string recommender via the recommender traits panel
* Go to the annotations screen and observe suggestions
* Delete gazeteer (to see that this also works)
* Export/re-import a project with Gazeteers and see if they are still there and work in the re-imported project

**Automatic testing**

**Documentation**
